### PR TITLE
1 level

### PR DESCRIPTION
**Pull request by Kropisha**

### CODE REVIEW

File: WebApi/Controller/LibraryController.cs [hunk 1]
```diff
@@ -10,38 +10,66 @@ namespace WebApi.Controller
     using WebApi.Models;
     [Route("api/[controller]")]
 
+    /// <summary>
+    /// Library class
```
Comment (FSuhov) [inline WebApi/Controller/LibraryController.cs]: It is not Library class. It is controller, that works on requests and responds to it.

File: WebApi/Controller/LibraryController.cs [hunk 2]
```diff
@@ -10,38 +10,66 @@ namespace WebApi.Controller
...
         }
 
+        /// <summary>
+        /// Edit exists book
+        /// </summary>
+        /// <param name="id">book's id</param>
+        /// <param name="book">book's instance</param>
         [Route("api/library/edit/{id},{book}")]
         [HttpPut]
         public void EditBook(int id, Book book)
```
Comment (FSuhov) [inline WebApi/Controller/LibraryController.cs:72]: I might be wrong, I but probably according to standards it should be UpdateBook. Not Edit

File: WebApi/Library/BooksSet.cs [hunk 3]
```diff
@@ -22,23 +31,41 @@ public BooksSet()
...
         }
 
+        /// <summary>
+        /// Get one book
+        /// </summary>
+        /// <param name="id">book`s id</param>
+        /// <returns>one book</returns>
         public Book GetBook(int id)
         {
             Book book = this.library.ElementAt(id); // Find
```
Comment (FSuhov) [inline WebApi/Library/BooksSet.cs]: I am afraid this is wrong. You need to return the book with the specified Id, not at the position of Id.
Please consider the following: this.library.FirstOrDefault(b => b.Id == id);

File: WebApi/Library/BooksSet.cs [hunk 4]
```diff
@@ -22,23 +31,41 @@ public BooksSet()
...
             return book;
         }
 
+        /// <summary>
+        /// For creating books
+        /// </summary>
+        /// <param name="book">book instance</param>
         [HttpPost]
```
Comment (FSuhov) [inline WebApi/Library/BooksSet.cs]: As I understand, it is a business logic class. It does not anything about Http requests. Therefore, no need to use [HttpPost] attribute here.

File: WebApi/Library/BooksSet.cs [hunk 5]
```diff
@@ -22,23 +31,41 @@ public BooksSet()
...
         {
             this.library.Add(book);
         }
 
+        /// <summary>
+        /// Edit exist book
+        /// </summary>
+        /// <param name="id">book's id</param>
+        /// <param name="book">book's instance</param>
         [HttpPut]
```
Comment (FSuhov) [inline WebApi/Library/BooksSet.cs]: Same as previous comment

File: WebApi/Models/IBook.cs [hunk 6]
```diff
@@ -9,12 +9,24 @@ namespace WebApi.Models
     /// </summary>
```
Comment (FSuhov) [inline WebApi/Models/IBook.cs]: I might be wrong again, but I believe, this interface is not needed here. You need to create interface for BookSet instead.

File: WebApi/Controller/LibraryController.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+// Copyright (c) 2018 All Rights Reserved
+// </copyright>
+// <author>Yuliia Kropyvna</author>
+namespace WebApi.Controller
+{
+    using System.Collections.Generic;
+    using System.Web.Http;
+    using WebApi.Library;
+    using WebApi.Models;
+    [Route("api/[controller]")]
```
Comment (anton-borodulin-softserve) [inline WebApi/Controller/LibraryController.cs:11]: Attributes should be placed near the place of usage.  Here at the line 15.

File: WebApi/Library/BooksSet.cs [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        /// <summary>
+        /// Get one book
+        /// </summary>
+        /// <param name="id">book`s id</param>
+        /// <returns>one book</returns>
+        public Book GetBook(int id)
+        {
+            Book book = this.library[id - 1];
```
Comment (anton-borodulin-softserve) [inline WebApi/Library/BooksSet.cs:68]: Book Id and book index is different things. Yeah in your case they are almost the same but please change that for level2 PR. Index and Id shouldn't be related.

File: WebApi/Configuration/WebApiConfig.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
+﻿// <copyright file="WebApiConfig.cs" company="My Company Name">
+// Copyright (c) 2018 All Rights Reserved
+// </copyright>
+// <author>Yuliia Kropyvna</author>
+namespace WebApi.Configuration
+{
+    using System.Web.Http;
```
Comment (anton-borodulin-softserve) [inline WebApi/Configuration/WebApiConfig.cs:7]: Please, for level 2 recreate solution using .net core. not the .net framework.

**Review by anton-borodulin-softserve — APPROVED** [review]
Please proceed with level 2 as fast as you can. And I want to ask you to review level3 code for Denis and Maksim, I already approved them. I hope if you will check their code you will able to avoid mistakes

**Pull request closed without merging**